### DISCUSSION TITLE
Extend setting base transform to withTransforms param

### DISCUSSION THREAD

**Discussion by nixondesigndev** [craftcms/cms#8862] [Ideas]
### Description

This is a request to add support for specifying a base transform (https://github.com/craftcms/cms/issues/5853#issuecomment-652692013) when eager loading asset transforms with `withTransforms`.

Eager loading image transforms has significantly reduced our db queries (big thanks for that). We're now experimenting adding support for webp using the picture tag but can foresee issues with the implementation of https://github.com/craftcms/cms/issues/6381 as I'd assume Craft will need to do a whole new set of queries to do the overrides.

Perhaps eager loading could be enabled with a `withTransformOverrides` param or `withTransforms` takes an object of variants.

```
.withTransforms([{width: 300, height: 300}, {sizes: ['1.5x', '2x', '3x'], format: ['webp']})
```

**Comment by brandonkelly** [comment]
That `$transformOverrideParameters` argument was removed in c4696c68064238a32d5bf53bd9249abbdd7c8c63, so just updated your issue.
